packages/@cdktf/hcl2cdk/test/expressionToTs.test.ts

```diff
@@ -417,7 +417,7 @@ describe("expressionToTs", () => {
       getType
     );
     expect(code(result)).toMatchInlineSnapshot(
-      `""\${" + awsS3BucketExamplebucket.fqn + "}.network_interface[0].access_config[0].assigned_nat_ip""`
+      `"cdktf.Token.asString(cdktf.propertyAccess(awsS3BucketExamplebucket, ["network_interface", "0", "access_config", "0", "assigned_nat_ip"]))"`
```

Comment:
I think this would be a nicer goal to target, making everything possible access via TS and only then fall back to property access. What do you think, is this out of scope?

```suggestion
    `"cdktf.Token.asString(cdktf.propertyAccess(awsS3BucketExamplebucket.networkInterface, ["0", "access_config", "0", "assigned_nat_ip"]))"`
```

Reply:
Fixed now.